apis/core/v1alpha2/core_types.go

```diff
@@ -52,7 +51,7 @@ type WorkloadDefinitionSpec struct {
 
 	// Extension is used for extension needs by OAM platform builders
 	// +optional
-	Extension *unstructured.Unstructured `json:"extension,omitempty"`
+	Extension *runtime.RawExtension `json:"extension,omitempty"`
```

Comment:
I think we should use runtime.RawExtension instead of a pointer, that might fix the build

Reply:
@ryanzhang-oss Yes, it could fix the issue as below.
```
v1alpha2.TraitDefinitionList.ListMeta: v1.ListMeta.TypeMeta: Kind: Items: []v1alpha2.TraitDefinition: v1alpha2.TraitDefinition.Spec: v1alpha2.TraitDefinitionSpec.Extension: unmarshalerDecoder: Object 'Kind' is missing in '{"alias":"ManualScaler","object":{"apiVersion":"core.oam.dev/v1alpha2","kind":"ManualScalerTrait","spec":{"replicaCount":2}},"parameters":[{"default":"5","fieldPaths":["spec.replicaCount"],"name":"replicaCount","required":true,"type":"int"}]}', error found in #10 byte of ...|":"int"}]}}}],"kind"|..., bigger context ...|me":"replicaCount","required":true,"type":"int"}]}}}],"kind":"TraitDefinitionList","metadata":{"cont|...
```

But it will failed the e2e-test. There is one test case of applying a simple traitdefinition without spec.extension which could be empty, it will hit the issue as below.
>/Users/zhouzhengxi/Programming/golang/src/github.com/crossplane/oam-kubernetes-runtime/test/e2e-test/suite_test.go:59

>  Expected
      <*errors.StatusError | 0xc00026c1e0>: {
          ErrStatus: {
              TypeMeta: {Kind: "", APIVersion: ""},
              ListMeta: {
                  SelfLink: "",
                  ResourceVersion: "",
                  Continue: "",
                  RemainingItemCount: nil,
              },
              Status: "Failure",
              Message: "TraitDefinition.core.oam.dev \"manualscalertraits.core.oam.dev\" is invalid: spec.extension: Invalid value: \"null\": spec.extension in body must be of type object: \"null\"",
              Reason: "Invalid",
              Details: {
                  Name: "manualscalertraits.core.oam.dev",
                  Group: "core.oam.dev",
                  Kind: "TraitDefinition",
                  UID: "",
                  Causes: [
                      {
                          Type: "FieldValueInvalid",
                          Message: "Invalid value: \"null\": spec.extension in body must be of type object: \"null\"",
                          Field: "spec.extension",
                      },
                  ],
                  RetryAfterSeconds: 0,
              },
              Code: 422,
          },
      }
  To satisfy at least one of these matchers: [%!s(*matchers.BeNilMatcher=&{}) %!s(*util.AlreadyExistMatcher=&{})]